### PR TITLE
Closes 3878 - refactors rounding functions to new interface, pulls hash function into their own procs

### PR DESCRIPTION
**Pull request by drculhane**

### CODE REVIEW

**Review by ajpotts — APPROVED** [review]
This is great!  Thanks Andy!

File: tests/numpy/numeric_test.py [hunk 1]
```diff
@@ -584,19 +586,18 @@ def test_isnan(self):
...
-        nda = np.array([0,9999.9999])
+        nda = np.array([0, 9999.9999])
         pda = ak.array(nda)
         warnings.filterwarnings("ignore")
         nda_blowup = np.exp(nda)
         warnings.filterwarnings("default")
         pda_blowup = ak.exp(pda)
         assert (np.isinf(nda_blowup) == ak.isinf(pda_blowup).to_ndarray()).all()
         assert (np.isfinite(nda_blowup) == ak.isfinite(pda_blowup).to_ndarray()).all()
-
```
[the remark this answers is not in the document]
Comment (drculhane) [reply tests/numpy/numeric_test.py:626]: Done.

File: src/EfuncMsg.chpl [hunk 2]
```diff
@@ -140,6 +140,71 @@ module EfuncMsg
...
+            (a1i, a2i) = sipHash128(x): (uint, uint);
+        }
+        var repMsg = "created " + st.attrib(rname2) + "+";
+        repMsg += "created " + st.attrib(rname);
+        eLogger.debug(getModuleName(),getRoutineName(),getLineNumber(),repMsg);
+        return new MsgTuple(repMsg, MsgType.NORMAL);
+    }
+    proc hash128 (cmd: string, msgArgs: borrowed MessageArgs, st: borrowed SymTab, type array_dtype, param array_nd: int) : MsgTuple throws {
+        return MsgTuple.error("hash128 does not support type %s".format(types2str(array_dtype)));
+    }
```
[the remark this answers is not in the document]
Comment (drculhane) [reply src/EfuncMsg.chpl]: Done :)

[the remark this answers is not in the document]
Comment (ajpotts) [reply src/EfuncMsg.chpl]: Thanks!

**Review by jaketrookman — APPROVED** [review]
Looks good

**Pull request merged**